### PR TITLE
Add @primer/octicons

### PR DESCRIPTION
**Pull request by olivermrose**
Please fill in this template.

- [x] Use a meaningful title for the pull request. Include the name of the package modified.
- [x] Test the change in your own code. (Compile and run.)
- [x] [Add or edit tests](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#my-package-teststs) to reflect the change.
- [x] Follow the advice from the [readme](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#make-a-pull-request).
- [x] Avoid [common mistakes](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#common-mistakes).
- [x] [Run `npm test <package to test>`](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#running-tests).

Select one of these and delete the others:

If adding a new definition:
- [x] The package does not already provide its own types, or cannot have its `.d.ts` files generated via `--declaration`
- [x] If this is for an npm package, match the name. If not, do not conflict with the name of an npm package.
- [x] Create it with `dts-gen --dt`, not by basing it on an existing project.
- [x] Represents shape of module/library [correctly](https://www.typescriptlang.org/docs/handbook/declaration-files/library-structures.html)
- [x] `tslint.json` [should contain](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#linter-tslintjson) `{ "extends": "dtslint/dt.json" }`, and no additional rules.
- [x] `tsconfig.json` [should have](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#tsconfigjson) `noImplicitAny`, `noImplicitThis`, `strictNullChecks`, and `strictFunctionTypes` set to `true`.

### CODE REVIEW

File: types/primer__octicons/index.d.ts [hunk 1]
```diff
@@ -0,0 +1,333 @@
...
+        /**
+         * This is an object of all the attributes that will be added to the output tag.
+         */
+        readonly options: IconAttributes;
+    }
+
+    interface IconAttributes {
+        version: string;
+        width: string;
+        height: string;
```
Comment (peterblazejewicz) [inline types/primer__octicons/index.d.ts:61]:
```suggestion
        height: number;
```
https://github.com/primer/octicons/blob/main/lib/octicons_node/index.js#L55
(assuming is succeds)

File: types/primer__octicons/index.d.ts [hunk 2]
```diff
@@ -0,0 +1,333 @@
...
+
+        /**
+         * This is an object of all the attributes that will be added to the output tag.
+         */
+        readonly options: IconAttributes;
+    }
+
+    interface IconAttributes {
+        version: string;
+        width: string;
```
Comment (peterblazejewicz) [inline types/primer__octicons/index.d.ts:60]:
```suggestion
        width: number;
```
assuming it comes from JSON:
```json
"width": 12,
"height": 16,
```

**Review by peterblazejewicz — APPROVED** [review]
LGMT!
@vBread thx!

**Comment by vBread** [comment]
Ready to merge

**Pull request merged**